mteb/encoder_interface.py

```diff
@@ -151,3 +156,56 @@ def convert_conv_history_to_query(conversations: Sequence[Sequence[str]]) -> str
             The query.
         """
         ...
+
+
+class ImageEncoder:
+    """Interface for image encoder designed based on VLM2VecWrapper.
+    There is not a perfect 1-1 match, e.g. device can be None here.
+    The intention here is to define the current interface and adapt to as close to MTEB as possible
+    and align as much as possible with sentencetransformers.
+    """
+
+    def __init__(
+        self,
+        device: str | None,
+        **kwargs: Any,
+    ):
+        pass
+
+    def encode(  # current a 1-1 match with Encoder.encode
+        self,
+        sentences: Sequence[str],
+        *,
+        task_name: str,
+        prompt_type: PromptType | None = None,
+        **kwargs: Any,
+    ) -> np.ndarray:
+        pass
+
+    def get_image_embeddings( # Seems like sentence transformers use a singular encode for both images and text. Not sure if we want to do the same.
```

Comment:
I have asked about the sentenceTransformer.encode interface for images here:

https://github.com/UKPLab/sentence-transformers/issues/3118

Reply:
Hmm if a singular encode is used for both images and text, then type checking is needed? Would that be sort of like the "fused_embedding" method with a bunch of if-statements handling the presence of types?

Reply:
I think so yes - I also discussed it a bit with Tom Aarsen on slack, seems to be the direction (sentence) transformers are going as well with one "general processing" component (e.g. a tokenizer) followed by a transformer encoders.

The exact input format (document in this case) is a bit unclear.

Reply:
having a general `encode` sounds good. The encoding of different modalities can happen behind the scenes similar to `get_fused_embedding` where it can take in an image and a text or either.